vshard/storage/init.lua

```diff
@@ -1137,6 +1137,7 @@ local function recovery_step_by_type(type)
     local total = 0
     local start_format = 'Starting %s buckets recovery step'
     for _, bucket in _bucket.index.status:pairs(type) do
+        lfiber.testcancel()
```

Comment:
I did consider exactly this at first. But then realized, that it would look worse.

- We already have `util.fiber_cond_wait()`. It exists exactly to avoid any exceptions. No-exceptions is often needed to perform proper cleanup in case of an error.

- Given the previous point we would need to give the new functions weird names like `util.fiber_cond_wait_or_cancel_xc()`. We can't call them just `util.fiber_cond_wait_xc()`, because it would imply it can throw also on timeout.

- There is `bucket_generation_wait()` which has cond wait inside. I would have to rename it also to `..._or_cancel_xc()`.

I figured the current approach is more explicit and won't look as ugly.